### ISSUE TITLE
Strange training dynamics for ImageReward model.

### ISSUE THREAD

**Issue by bhattg** [THUDM/ImageReward#64]
Hi! I am trying to train a reward model, and I am confused why in the initial iterations of training the gradients are not changing and neither the loss is changing. Only after some steps does it suddenly change and then learning is completed. 

Following is the attached learning dynamics. 
![Screen Shot 2023-10-08 at 6 21 46 PM](https://github.com/THUDM/ImageReward/assets/39345600/f8eedd9f-2ce8-41cb-b474-4ef0ba635272)

**Comment by learn01one** [comment]
Hello, which version of python and cuda are you using? Thank you.

**Comment by xujz18** [comment]
This is a very interesting discovery, and I believe it may be related to the learning rate schedule and warm-up settings, although there could be other factors worth exploring.

**Comment by bhattg** [comment]
Hello, sorry I couldn't get back with the question on python version 3.10.13 and CUDA 11.7

Experiment was run using torch 1.13.0 

Regarding the learning dynamics, I am using the following 

```--fix_rate 0.7 --lr 1e-05 --lr-decay-style cosine --warmup 0.0  --batch_size 32 --accumulation_steps 1 --epochs 50 ```
